src/App.tsx

```diff
@@ -1,27 +1,82 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+enum Sort {
+  All = 'all-button',
+  FirstFive = 'first-five-button',
+  Red = 'red-button',
+}
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+const getButtonName = (sortType: Sort) => {
+  switch (sortType) {
+    case Sort.All:
+      return 'Load all goods';
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    case Sort.FirstFive:
+      return 'Load 5 first goods';
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    case Sort.Red:
+      return 'Load red goods';
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    default:
+      return '';
+  }
+};
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [sort, setSort] = useState<Sort>();
+
+  const getGoods = useCallback(async (promise: Promise<Good[]>) => {
+    setGoods(await promise);
```

Comment:
await, cool 👍 
but what if there will be an error?
and how about users with slow internet connection? I guess they can be a bit confused by fact that nothing is happen on the page when they click `load` button 🙂 

We as developers that working on product improving should keep users with us and should consider users needs, don't we? 😉